File: 240327/softeer-6270.cpp
#include<iostream>
#include<vector>

using namespace std;

int main(int argc, char** argv)
{
    int N, M;
    cin >> N >> M;

    vector<int> standard = {0};
    for(int i=0; i<N; i++){
        int length, speed;
        cin >> length >> speed;
        for(int j=0; j<length; j++){
            standard.push_back(speed);
        }
    }

    int answer = 0;
    int length_start = 1;
    for(int i=0; i<M; i++){
        int length, speed;
        cin >> length >> speed;
        for(int j=length_start; j<(length_start+length); j++){
            answer = max(answer, speed - standard[j]);
        }
        length_start += length;
    }
    cout << answer;
    return 0;
}

File: 240327/softeer-6270.py
N, M = map(int, input().split())
standard = [0]
for _ in range(N):
    length, speed = map(int, input().split())
    for _ in range(length):
        standard.append(speed)

answer = 0
length_start = 1
for _ in range(M):
    length, speed = map(int, input().split())
    for l in range(length_start, length_start+length):
        answer = max(answer, speed - standard[l])
    length_start += length

print(answer)